nmigen/lib/io.py

```diff
@@ -45,3 +45,48 @@ def elaborate(self, platform):
         f = m.elaborate(platform)
         f.flatten = True
         return f
+
+
+class DiffInput:
+    def __init__(self, p, n, i):
+        self.p = p
+        self.n = n
+        self.i = i
+
+    def elaborate(self, platform):
+        m = Module()
```

Comment:
The entire body of this function can be replaced with `platform.get_diff_input(self.p, self.n, self.i)`, it is a no-op as written.

But also I think we don't actually need any of these classes. You either:
  * request a prepared pad (SDR, DDR, whatever) from the platform, in which case it gives you a Record already connected to an Instance, or
  * request a raw pad from the platform, in which case it gives you a bare inout.

In both cases we don't need `DiffInput`, etc, only `pad_layout`.